mydata.R

```diff
@@ -23,16 +21,35 @@ get_data_fromDB<-function(credentials=credentials, sqlquery){
 
 }
 
+get_seasons <- function(df) {
+  'adds a seasons column to the data according to the south african seasons'
+  df1$season <- df1$month
+  
+  df1[df1$season=="Dec"| df1$season=="Jan"| df1$season== "Feb","season"] <- "summer"
+  df1[df1$season=="Mar"| df1$season=="Apr"| df1$season== "May","season"] <- "autumn"
+  df1[df1$season=="Jun"| df1$season=="Jul"| df1$season== "Aug","season"] <- "winter"
+  df1[df1$season=="Sep"| df1$season=="Oct"| df1$season== "Nov","season"] <- "spring"
+  
+  return(df1)
+}
```

Comment:
There are some nifty functions that dplyr gives us for this use case :)

```
df %>%
  mutate(season = case_when(month %in% c("Dec", "Jan", "Feb") ~ "summer",
                            month %in% c("Mar", "Apr", "May") ~ "atumn",
                            month %in% c("Jun","Jul","Aug") ~ "winter",
                            month %in% c("Sep","Oct","Nov") ~ "spring"))
```
and then don't forget to make sure that the column is categorized using `factor`, instead of a character column.